analytics/models.py

```diff
@@ -30,4 +35,56 @@ class submission_analytics(models.Model):
     difficulty = models.DecimalField(max_digits=8, decimal_places=4, default=0)
     effort = models.DecimalField(max_digits=8, decimal_places=4, default=0)
     error = models.DecimalField(max_digits=8, decimal_places=4, default=0)
-    test_time = models.DecimalField(max_digits=8, decimal_places=4, default=0)
\ No newline at end of file
+    test_time = models.DecimalField(max_digits=8, decimal_places=4, default=0)
+
+
+class QuestionAnalytics(models.Model):
+    id = models.AutoField(primary_key=True)
+    question = models.ForeignKey(Question, on_delete=models.CASCADE)
+    event = models.ForeignKey(Event, on_delete=models.CASCADE, null=True)
+    number_submission = models.IntegerField(default=0)
+    avg_grade = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    correct_rate = models.DecimalField(max_digits=8, decimal_places=4, default=0)
+
+    frequently_wrong_ans = models.CharField(max_length=5, default="n/a")
+    frequently_wrong_reason = models.CharField(max_length=255, default="n/a")
+
+    lines = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    blank_lines = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    comment_lines = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    import_lines = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    cc = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    method = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    operator = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    operand = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    unique_operator = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    unique_operand = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    vocab = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    size = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    vol = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    difficulty = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    effort = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    error = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+    test_time = models.DecimalField(max_digits=8, decimal_places=2, default=0)
+
+
+# class EventAnalytics(models.Model):
```

Comment:
Don't include commented code in the PRs.

Reply:
Removed.